### ISSUE TITLE
7$: improve diagnostic for 'const C = abstract class {}'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#850]
In the following TypeScript code, the user intended to create an abstract class using expression syntax (which is not valid TypeScript):
```typescript
const C = abstract class {}
```

We produce a bad diagnostic. Produce a helpful diagnostic instead.
